iree/base/time.c

```diff
@@ -33,6 +33,9 @@ IREE_API_EXPORT iree_time_t iree_time_now(void) {
   struct timespec clock_time;
   clock_gettime(CLOCK_REALTIME, &clock_time);
   return clock_time.tv_nsec;
+#elif defined(IREE_PLATFORM_GENERIC) && defined(IREE_ARCH_RISCV_32)
```

Comment:
We need this for `IREE_ARCH_ARM_32` as well. Do we explicitly want to check for all those potential bare-metal targets or would it be enough to just check if `IREE_PLATFORM_GENERIC` is defined?

Reply:
`IREE_PLATFORM_GENERIC` doesn't necessarily mean there's no clock mechanism in the system. The no-op code path could potentially be problematic (see https://github.com/google/iree/issues/6027#issuecomment-854104208). If the time mechanism is well-defined then it should be plugged in there instead of no-op IMHO.

Reply:
I see... However, I think we may want to define an additional flag or to implement some other mechanism instead of relying on the architecture. E.g. for Coretex-M4 it depends on the firmware, as you can build upon [libopencm3](https://github.com/libopencm3/libopencm3) or on STM32Cube with [X-CUBE-RTC](https://www.st.com/en/embedded-software/x-cube-rtc.htm). As it isn't clear which firmware is going to be used, we're going for a no-op in the first step and we should somehow allow users to plugin a timing mechanism more easily. @dajuro FYI.

Reply:
we can use IREE_USER_CONFIG_H for things that vary a ton - defining an optional IREE_TIME_NOW_FN that #ifdef would skip these implementations would let custom targets choose whatever they wanted to call

Reply:
(including `#define IREE_TIME_NOW_FN { return 0; }` for a no-op)

Reply:
I am not sure if I follow the logic. If we set in `iree/base/config.h`

```
#if !defined(IREE_TIME_NOW_FN)
#define IREE_TIME_NOW_FN { return 0; }
#endif  
```

and in `iree/base/time.c`
```
#elif defined(IREE_PLATFORM_GENERIC)
  IREE_TIME_NOW_FN
...
```
Then for all generic platform it will always be a no-op unless user configure it differently. This would work, but I don't see how #ifdef come into play.